lib/elixir/src/elixir_translator.erl

```diff
@@ -9,12 +9,7 @@
 %% =
 
 translate({'=', Meta, [Left, Right]}, S) ->
-  Return = case Left of
-    {'_', _, Atom} when is_atom(Atom) -> false;
```

Comment:
So this variable tells whether the return value matters or not. If we are assigning to a variable which is not _ the return value definitely matters. However, if we are assigned to `_`, the return value should matter only if the parent scope told us if it matters or not. So I would say the fix is replacing the `false` above by `S#elixir_scope.return`.